src/foam/nanos/auth/UserAndGroupAuthService.java

```diff
@@ -167,6 +167,46 @@ public User loginByEmail(X x, String email, String password) throws Authenticati
     return user;
   }
 
+  /**
+    Checks if the user passed into the method has the passed
+    in permission attributed to it by checking their group.
+    No check on User and group enabled flags.
+  */
+  public Boolean checkUserPermission(foam.core.X x, User user, String permission) {
+    // check whether user has permission to check user permissions
+    if ( ! check(x, permissionCheckPrivilege_) ) {
```

Comment:
Where is permissionCheckPrivilege defined? If it's a constant, it should be capitalized.